apps/auxo-app/src/store/vault/vault.slice.ts

```diff
@@ -1,90 +1,196 @@
 import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 import { vaultState } from './vault.state';
-import { Balance, UserBalanceOnChainData, Vault, VaultAuth, VaultOnChainData } from './Vault';
-import { toBalance } from '../../utils';
-import { addBalances, subBalances } from '../../utils/balances';
+import { Balance, Vault } from './Vault';
+import { convertFromUnderlying, convertToUnderlying } from '../../utils';
+import { addBalances, subBalances, zeroBalance } from '../../utils/balances';
+import { thunkApproveDeposit, thunkMakeDeposit, thunkConfirmWithdrawal, thunkIncreaseWithdrawal, thunkAuthorizeDepositor } from './vault.thunks';
 
 export const vaultSlice = createSlice({
   name: 'vault',
   initialState: vaultState,
+
+  /**
+   * Extra reducers here subscribe to the async thunks 'fulfilled' hook
+   * and prepare the state transformations for the corresponding vault.
+   * To save repeating the find and update logic, we re-use the caseReducer 
+   * `setVault`, which finds and updates a single vault, according to
+   * the transformer function.
+   */
```

Comment:
Note, these are where we can define optimistic UI updates. Eventual consistency will be reached when the onChain data is loaded